adoc/chapters/programming_interface.adoc

```diff
@@ -10089,11 +10108,17 @@ a@ Pointer is suitably aligned for an object of type [code]#T# or it is aligned
 
 ==== {cpp} allocator interface
 
-USM provides an allocator class that aligns with the {cpp} allocator interface.  This is useful when
-using USM with things like {cpp} containers or routines like [code]#std::allocate_shared#.  However, due
-to how {cpp} library routines that use allocators tend to assume that any memory allocated by an allocator
-may be accessed on the host, the [code]#usm_allocator# class is not required to support
-allocating [code]#usm::alloc::device# allocations.
+SYCL defines an allocator class named [code]#usm_allocator# that satisfies the
+{cpp} named requirement [code]#Allocator#.  The [code]#AllocKind# template
+parameter can be either [code]#usm::alloc::host# or [code]#usm::alloc::shared#,
+causing the allocator to allocate either USM host or USM shared memory.
```

Comment:
I think we need some term because there are 3 functions that could allocate "host USM" memory as well as the `usm_allocator`.  It would get very verbose to say "Unified Shared Memory allocation returned by `malloc_host`, `aligned_alloc_host`, `sycl::malloc` (when called with `usm::alloc::host`), or the `usm_allocator` (when instantiated with `usm::alloc::host`)"

I don't have a preference between "USM host allocations" vs. "host USM allocations".  If we use the latter already in the OpenCL spec, then let's standardize on that.

I think this term will be more clearly specified once we merge the remaining parts of the clarifications I want to do.  Note that the internal PR #8 adds new sub-sections named "USM device allocations", "USM host allocations", and "USM shared allocations".  I'll change the word order in #8 to say "Device USM allocations", etc.

Do you have any concern if I continue to use the phrase "host USM memory" as a shorthand for "memory in a host USM allocation"?

Reply:
> Do you have any concern if I continue to use the phrase "host USM memory" as a shorthand for "memory in a host USM allocation"?

No concern, this sounds good, and I also prefer "host USM" over "USM host".

I have a slight preference for "host USM allocations" or even just "host USM" instead of "host USM memory", just because the M in USM stands for "memory", but that's awfully pedantic.

Reply:
I reversed the word order in terms like "USM host allocation" to be "host USM allocation".  I also changed occurrences like "USM memory" to avoid the redundant "memory".  These were done in bcff4184af20631985102c25dfb6fc3e656225f0